auth0/src/test/java/com/auth0/android/request/internal/BaseRequestTest.kt

```diff
@@ -400,56 +450,143 @@ public class BaseRequestTest {
         )
     }
 
+    /**
+     * @return Function indicating whether the response body `InputStream` was closed
+     */
     @Throws(Exception::class)
-    private fun mockSuccessfulServerResponse() {
+    private fun mockSuccessfulServerResponse(): () -> Boolean {
```

Comment:
I'm not 100% convinced about the signature changes. Now, these functions return a function that would yield a boolean, but the name of each of the 3 functions is not implying that. Do you think it could be refactored? Maybe make it return something else (i.e. the input stream) and wrap that with a separate function to check if it was closed? 

What are your thoughts?

Reply:
You are right, the function now becomes responsible for two tasks which is probably not a good design. But I am not sure if there is an easier solution which does not make the tests much more verbose.
Returning the `AwareInputStream` might have the same problem because the functions are not really producing the input stream as mocked response, but instead are modifying the mocks, so returning the input stream might be equally irritating.

Maybe an alternative would be to add a mutable `wasResponseBodyClosed: Boolean` property to the test class? What do you think? The test functions not using the mocking functions could just ignore that property.

Reply:
@Marcono1234 Yes, I think that would be better. Not that I don't like the kotlin way shown in the current solution, but the double-task part is not convincing me. As long as we reset the value to "false" before each test, adding a property should be ok.

Alright, those should be the last set of changes. Thanks for sticking to this PR!